client/src/app/reducers/__tests__/items-entities-reducer.test.js

```diff
@@ -0,0 +1,70 @@
+import { render, screen } from "@testing-library/react";
+import itemsEntitiesReducer from "../items-entities-reducer";
+
+const initialState = {
+  items: {},
+};
+
+describe("items Entities Reducer ", () => {
+  const prevState = { ...initialState };
+  let newStateAfterSetData;
+  test("return prev state when no match case for a given action", () => {
+    const action = { type: "ADD_USER" };
+    const newState = itemsEntitiesReducer(prevState, action);
+    expect(newState).toEqual(prevState);
+  });
+
+  test("return new state with 3 new items", () => {
+    const data = [
+      { id: 676, itemName: "Go to walk", status: false },
+      { id: 677, itemName: "Eat some candy", status: true },
+      { id: 678, itemName: "Drive to LA", status: false },
+    ];
+    const action = { type: "SET_DATA", data };
+    newStateAfterSetData = itemsEntitiesReducer(prevState, action);
+    expect(Object.keys(newStateAfterSetData.items).length).toEqual(3);
+  });
+
+  test("return new state without item with id:677 ", () => {
+    const action = { type: "REMOVE_ITEM", id: 677 };
+    const newStateAfterRemoveItem = itemsEntitiesReducer(
+      newStateAfterSetData,
```

Comment:
you shouldn't rely on data from the previous test in a unit test.
in this case if you will remove the above test all the other will fail because "newStateAfterSetData" will always be null